proton

```diff
@@ -165,9 +168,10 @@ class CompatData:
                         dirs.append(path)
             for d in dirs:
                 try:
-                    os.rmdir(d)
-                except OSError:
-                    #not empty
+                    if not any(os.scandir(d)):
+                        os.rmdir(d)
+                except OSError as err:
```

Comment:
Instead
```
except OSError as e:
    if e.errno == errno.ENOTEMPTY:
      ...
```

Reply:
As far as I could gather checking if the directory is empty before attempting to do the operation is less expensive than letting it throw an exception and catching it. Would we still prefer going with the latter approach?

Reply:
There's no measurable performance difference between the two (although the deletion and seeing whether it fails is most likely faster as it results in less syscalls) but with the exception handling race conditions are impossible. That's typically recommended way to handle these things.

Reply:
Makes sense. I hadn't considered race conditions. I'll make the changes in a bit :)